src/Microsoft.VisualStudio.Services.Agent/ProcessChannel.cs

```diff
@@ -50,15 +50,18 @@ public sealed class ProcessChannel : AgentService, IProcessChannel
         private StreamString _writeStream;
         private StreamString _readStream;
 
-        public void StartServer(StartProcessDelegate startProcess)
+        public void StartServer(StartProcessDelegate startProcess, bool disposeClient = true)
         {
             _outServer = new AnonymousPipeServerStream(PipeDirection.Out, HandleInheritability.Inheritable);
             _inServer = new AnonymousPipeServerStream(PipeDirection.In, HandleInheritability.Inheritable);
             _readStream = new StreamString(_inServer);
             _writeStream = new StreamString(_outServer);
             startProcess(_outServer.GetClientHandleAsString(), _inServer.GetClientHandleAsString());
-            _outServer.DisposeLocalCopyOfClientHandle();
-            _inServer.DisposeLocalCopyOfClientHandle();
+            if (disposeClient)
```

Comment:
When do we not want to dispose?

Reply:
The tests act as a client and server in the same process. We need avoid disposing the client handle since handles are per-process.

These get disposed properly when they go out of scope (see `Dispose()` in this class)